pkg/api/server.go

```diff
@@ -21,43 +21,37 @@ type ServerConfig struct {
 	MemberPass       string `split_words:"true" default:"admin"`
 	PrepRepToken     string `split_words:"true" default:"admin"`
 	AdminToken       string `split_words:"true" default:"admin"`
-}
-
-type StorageConfig struct {
 	SheetsBucketName string `split_words:"true" default:"sheets"`
 	ClixBucketName   string `split_words:"true" default:"clix"`
 	TracksBucketName string `split_words:"true" default:"tracks"`
 	RedisNamespace   string `split_words:"true" default:"local"`
 }
 
-type Storage struct {
-	StorageConfig
+type Database struct {
+	ServerConfig
 	Parts  *parts.RedisParts
 	Sheets *storage.Bucket
 	Clix   *storage.Bucket
 	Tracks *storage.Bucket
 }
 
-func NewStorage(ctx context.Context, warehouse *storage.Warehouse, config StorageConfig) *Storage {
+func NewServer(ctx context.Context, config ServerConfig) *http.Server {
 	var newBucket = func(ctx context.Context, bucketName string) *storage.Bucket {
-		bucket, err := warehouse.NewBucket(ctx, bucketName)
+		bucket, err := storage.NewBucket(ctx, bucketName)
 		if err != nil {
 			logger.WithError(err).WithField("bucket_name", bucketName).Fatal("warehouse.NewBucket() failed")
 		}
 		return bucket
 	}
 
-	db := Storage{
-		StorageConfig: config,
-		Sheets:        newBucket(ctx, config.SheetsBucketName),
-		Clix:          newBucket(ctx, config.ClixBucketName),
-		Tracks:        newBucket(ctx, config.TracksBucketName),
-		Parts:         parts.NewParts(config.RedisNamespace + ":parts"),
+	database := Database{
+		ServerConfig: config,
+		Sheets:       newBucket(ctx, config.SheetsBucketName),
+		Clix:         newBucket(ctx, config.ClixBucketName),
+		Tracks:       newBucket(ctx, config.TracksBucketName),
```

Comment:
These are all stored in the same bucket anyways... Probably no need to keep them in different pointers n such. Could just have 1 single `Distro` bucket.